src/components/Calculator.js

```diff
@@ -1,18 +1,33 @@
+import { useState } from 'react';
 import PropTypes from 'prop-types';
+import calculate from '../logic/calculate';
 import { BUTTON_VALUES, OPERATOR_VALUES } from '../constants';
 
-export const Calculator = () => (
-  <div className="calculator">
-    <div className="screen">
-      <Displaybar value={String(0)} />
-    </div>
-    <div className="touch-pad">
-      {BUTTON_VALUES.map((button) => (
-        <Button key={button} text={button} />
-      ))}
+export const Calculator = () => {
+  const [state, setState] = useState({});
+  const handleClick = (e) => {
+    const result = calculate({ ...state }, e.target.name);
+    setState((prev) => ({ ...prev, ...result }));
+  };
+  const { total, operation, next } = state;
+  return (
+    <div className="calculator">
+      <div className="screen">
+        <small>
+          {total}
+          {operation}
+          {next}
+        </small>
+        <Displaybar value={String(total || 0)} />
+      </div>
+      <div className="touch-pad">
+        {BUTTON_VALUES.map((button) => (
+          <Button key={button} text={button} handleClick={handleClick} />
```

Comment:
- [ ] Your division button does not work with the operate file. You need to use the correct symbol specified in the file, which is &div;, not /

  ![image](https://github.com/King-Benx/math-magician/assets/54848363/178783bc-23a4-446e-ac66-a51e5e418a7c)